Реализован функционал 8 Спринта

avralevm opened the pull request:
1.Добавлены поля  duration и startTime для всех видов Task 
2.Добавлен метод расчёта окончания задачи getEndTime()
3.Добавлен список Task отсортированный по полю startTime
4.Добавлен метод проверки пересечений isOverlapping
5.Добавлен класс TaskOverlapException
6.Заменены циклы на stream
7.Добавлены тесты
8.Добавлен базовый класс с тестами TaskManagerTest

avralevm committed:
Реализован функционал 8 Спринта

avralevm committed:
Исправил CodeStyle для прохождения тесто

avralevm committed:
Исправил CodeStyle для прохождения тесто

avralevm committed:
Исправил CodeStyle для прохождения тесто

avralevm committed:
Исправил CodeStyle для прохождения тесто

litvitnik commented on src/manager/InMemoryTaskManager.java:
```diff
@@ -198,7 +242,44 @@ private void updateStatus(Epic epic) {
         }
     }
 
+    private void updateEpicTimeFields(Epic epic) {
```
Этот метод должен находиться в классе Epic

avralevm replied:
Нормальная реализация будет если в метод передавать List<Subtask>?
Пример вызова метода: epic.updateEpicTimeFields(getAllSubtaskOfEpic(epic.getId())); 
Не могу придумать более лучшей реализации.

avralevm replied:
Стоит ли метод updateStatus тоже перенести в Epic, чтобы разгрузить класс InMemeryTaskManager или он обязательно должен быть private?

litvitnik replied:
Это не критично. Можешь сделать так, как тебе удобно)

litvitnik commented on src/manager/InMemoryTaskManager.java:
```diff
@@ -198,7 +242,44 @@ private void updateStatus(Epic epic) {
+                );
+    }
+
+    public List<Task> getPrioritizedTasks() {
```
Все публичные методы следует размещать в начале класса, а приватные методы должны находиться в конце класса

avralevm committed:
Скорректирован код с учётом комментариев:
1.Метод updateEpicTimeFields перенесён в Epic
2.Переделано хранение subtask в Epic(так как updateEpicTimeFields требует доступ как subtask)
3.В классе InMemoryTaskManager в методе removeAllEpics добавлено удаление всех Subtask из prioritizedTasks
4.Добвлены в класс TaskManagerTest 2 теста getHistory и getPrioritizedTasks
5.Все наименования тестов приведены к единому стандарту

avralevm committed:
метод updateStatus перенесён в Epic и переделан вызов его
